### PR TITLE
Fix FAR crash in REPL (#load'ed trees)...

### LINKED ISSUE #5244: NullReferenceException in LinkedFileReferenceFinder.DetermineCascadedSymbolsAsync for #load'ed Document
Create a "test.csx" file with the following content:
```
int i = 1;
```

In the C# Interactive Window, type the following (don't submit the last line):
```
> #load "test.csx"
> i
```

Move the cursor to the left of the "i" 

**RESULTS:**
```
NullReferenceException:
>	Microsoft.CodeAnalysis.Workspaces.dll!Microsoft.CodeAnalysis.FindSymbols.Finders.LinkedFileReferenceFinder.DetermineCascadedSymbolsAsync(Microsoft.CodeAnalysis.ISymbol symbol, Microsoft.CodeAnalysis.Solution solution, System.Collections.Immutable.IImmutableSet<Microsoft.CodeAnalysis.Project> projects, System.Threading.CancellationToken cancellationToken) Line 25	C#
 	Microsoft.CodeAnalysis.Workspaces.dll!Microsoft.CodeAnalysis.FindSymbols.FindReferencesSearchEngine.DetermineAllSymbolsCoreAsync.AnonymousMethod__0() Line 168	C#
```

### PR DESCRIPTION
**Pull request by KevinH-MS**
Fixes #5244

### CODE REVIEW

**Comment by KevinH-MS** [comment]
Alternatively, we could disable the ```IFindReferencesService``` in Interactive, but upon further consideration, that seems heavy-handed, and at least a couple people like the current highlight references experience.

**Comment by KevinH-MS** [comment]
FYI @amcasey @cston @tmat @ManishJayaswal @genlu

**Comment by tmat** [comment]
:+1:

**Comment by cston** [comment]
LGTM

**Comment by ManishJayaswal** [comment]
:+1:

File: src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/LinkedFileReferenceFinder.cs [hunk 1]
```diff
@@ -22,6 +19,15 @@ internal class LinkedFileReferenceFinder : IReferenceFinder
             {
                 var originalDocument = solution.GetDocument(location.SyntaxTree);
 
+                // GetDocument will return null for locations in #load'ed trees.
+                // TODO:  Remove this check and add logic to fetch the #load'ed tree's
+                // Document once https://github.com/dotnet/roslyn/issues/5260 is fixed.
+                if (originalDocument == null)
+                {
+                    Debug.Assert(solution.Workspace.Kind == "Interactive");
```
Comment (amcasey) [inline src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/LinkedFileReferenceFinder.cs]: I suspect the IDE team might prefer to make this part of the if condition and crash if its null in other cases.  @Pilchie?

**Comment by amcasey** [comment]
@dpoeschl for linked files.

**Comment by KevinH-MS** [comment]
I'm going to merge this, but @Pilchie @dpoeschl, if you'd rather see a Watson in the non-Interactive case, I can change the check.  My assumption is that this was simply an oversight, because most places that call ```GetDocument``` either do a null check or check for ```LocationKind.SourceFile``` (this wasn't doing either).

File: src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/LinkedFileReferenceFinder.cs (hunk 1 above)
Comment (Pilchie) [inline src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/LinkedFileReferenceFinder.cs]: Looking at callers, we have a mix.  @Mattwar, @jasonmalinowski - what do you expect here? That all syntax trees you can get to have documents?

Comment (mattwar) [inline src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/LinkedFileReferenceFinder.cs]: No, it should not be assumed that all SyntaxTree's have documents.  You should always guard against null and have an exit strategy.

**Pull request merged**